src/Kubernetes.Controller/Converters/YarpParser.cs

```diff
@@ -104,6 +104,24 @@ private static void HandleIngressRulePath(YarpIngressContext ingressContext, V1S
         }
     }
 
+    private static string UpstreamName(string namespaceName, V1IngressServiceBackend ingressServiceBackend)
+    {
+        if (ingressServiceBackend != null)
+        {
+            if (ingressServiceBackend.Port.Number.HasValue && ingressServiceBackend.Port.Number.Value > 0)
+            {
+                return $"{ingressServiceBackend.Name}.{namespaceName}:{ingressServiceBackend.Port.Number}";
+            }
+
+            if (!string.IsNullOrWhiteSpace(ingressServiceBackend.Name))
```

Comment:
Re-reading it, I think this was actually been the indented behavior?
```suggestion
            if (!string.IsNullOrWhiteSpace(ingressServiceBackend.Port.Name))
```

Reply:
@dpbevin Are you okay with me merging this with the above suggestion?